src/averageAges.js

```diff
@@ -37,7 +53,29 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let count = 0;
+
+  const womenArr = people.filter((person) => {
+    return person.sex === 'f';
+  });
+
+  const mothers = people.map(person => person.mother);
+
+  const sumAge = womenArr.reduce((prev, person) => {
+    const age = person.died - person.born;
+    const checkingChild = mothers.includes(person.name);
+    const checkingSex = ((checkingChild && withChildren) || !withChildren);
+
+    if (checkingSex) {
```

Comment:
Replace the `if` statement with a ternary operator for better readability. Example: `return checkingSex ? prev + age : prev;`